src/components/UptimeViewer/UptimeViewer.tsx

```diff
@@ -0,0 +1,77 @@
+import {DefinitionList} from '@gravity-ui/uikit';
+
+import {EMPTY_DATA_PLACEHOLDER} from '../../utils/constants';
+import {
+    formatDateTime,
+    getDowntimeFromDateFormatted,
+    getUptimeFromDateFormatted,
+} from '../../utils/dataFormatters/dataFormatters';
+import {CellWithPopover} from '../CellWithPopover/CellWithPopover';
+
+import i18n from './i18n';
+
+interface NodeUptimeProps {
+    StartTime?: string;
+    DisconnectTime?: string;
+}
+
+export function NodeUptime({StartTime, DisconnectTime}: NodeUptimeProps) {
+    let uptime: string | undefined;
+
+    if (DisconnectTime) {
+        uptime = getDowntimeFromDateFormatted(DisconnectTime);
+    } else {
+        uptime = getUptimeFromDateFormatted(StartTime);
+    }
+
+    if (!uptime) {
+        return EMPTY_DATA_PLACEHOLDER;
+    }
+    return (
+        <CellWithPopover
+            placement={['top', 'auto']}
+            content={
+                <DefinitionList responsive>
```

Comment:
should we add Popover if nor StartTime neither DisconnectTime is provided?

Reply:
Uptime won't be calculated in such case. I made the code more explicit